platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/DiagnosticToProblemListener.java

```diff
@@ -113,10 +113,19 @@ public void report(Diagnostic<? extends JavaFileObject> diagnostic) {
      *
      * @see com.sun.tools.javac.main.JavaCompiler#printCount(String, int)
      */
-    void printDiagnosticCounts() {
+    String diagnosticCounts() {
         Log logger = Log.instance(new Context());
-        printDiagnosticCount(logger, "error", errorCount);
-        printDiagnosticCount(logger, "warn", warningCount);
+        String error = diagnosticCount(logger, "error", errorCount);
+        String warning = diagnosticCount(logger, "warn", warningCount);
+        if (!"".equals(error) && !"".equals(warning)) {
+            return error + System.lineSeparator() + warning;
+        } else if ("".equals(error) && !"".equals(warning)) {
+            return warning;
+        } else if (!"".equals(error) && "".equals(warning)) {
+            return error;
+        } else {
+            return "";
+        }
```

Comment:
🤔 This is fine, but I would exploit the Streams API a bit to make this more readable

```suggestion
String error = diagnosticCount(logger, "error", errorCount);
String warning = diagnosticCount(logger, "warn", warningCount);

return Stream.of(error, warning)
        .filter(s -> !"".equals(s))
        .collect(Collectors.joining(System.lineSeparator()));
```

Reply:
🤔 Or if you would make `diagnosticCount` return null if the entry doesn't exists:

```suggestion
String error = diagnosticCount(logger, "error", errorCount);
String warning = diagnosticCount(logger, "warn", warningCount);

return Stream.of(error, warning)
        .filter(Objects::nonNull) // filter out null values
        .collect(Collectors.joining(System.lineSeparator()));
```